### PR TITLE
feat(mme): 5G Idle mode and service request procedure

### PR DESCRIPTION
**Pull request by VinayKumar1591**
5G Idle mode and service request procedure

       1. Tested on NGAP Tester
       2. Tested on DS Tester

Signed-off-by: vinay <vinay.tokala@radisys.com>

## Summary

Added changes to support Idle mode and service request procedure

## Test Plan

Tested on NGAP and DS Tester

### CODE REVIEW

**Review by ulaskozat — CHANGES_REQUESTED** [review]
Please add unit test coverage.

**Comment by VinayKumar1591** [comment]
> Please add unit test coverage.

Added UT cases

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GPDUSessionReActivationResult.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#include <cstring>
+#include "M5GCommonDefs.h"
+#include "M5GPDUSessionReActivationResult.h"
+
+using namespace std;
+namespace magma5g {
+M5GPDUSessionReActivationResult::M5GPDUSessionReActivationResult(){};
+M5GPDUSessionReActivationResult::~M5GPDUSessionReActivationResult(){};
+
+int M5GPDUSessionReActivationResult::EncodePDUSessionReActivationResult(
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GPDUSessionReActivationResult.cpp:24]: can you add unit test?

File: lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GPDUSessionReActivationResult.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    *(buffer + encoded) =
+        ((pduSessionReActivationStatus->pduSessionReActivationResult >> 8) &
+         0xFF);
+    encoded++;
+  }
+
+  return encoded;
+}
+
+int M5GPDUSessionReActivationResult::DecodePDUSessionReActivationResult(
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas5g/src/ies/M5GPDUSessionReActivationResult.cpp:45]: add unit test

File: lte/gateway/c/core/oai/tasks/nas5g/src/M5GServiceAccept.cpp [hunk 3]
```diff
@@ -54,6 +54,19 @@ int ServiceAcceptMsg::EncodeServiceAcceptMsg(
     return encoded_result;
   else
     encoded += encoded_result;
+  if ((encoded_result = svc_acpt->pdu_session_status.EncodePDUSessionStatus(
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas5g/src/M5GServiceAccept.cpp:57]: add unit test for EncodeServiceAcceptMsg

File: lte/gateway/c/core/oai/tasks/amf/amf_as.cpp [hunk 4]
```diff
@@ -488,6 +488,65 @@ int amf_reg_acceptmsg(const guti_m5_t* guti, amf_nas_message_t* nas_msg) {
...
+ **                                                                        **
+ ** Inputs:      msg:           The AMFMsg    primitive to process         **
+ **              Others:        None                                       **
+ **                                                                        **
+ ** Outputs:     amf_msg:       The AMF message to be sent                 **
+ **              Return:        The size of the AMF message                **
+ **              Others:        None                                       **
+ **                                                                        **
+ ***************************************************************************/
+static int amf_service_acceptmsg(
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/amf/amf_as.cpp:508]: add unit test

**Review by ulaskozat — APPROVED** [review]
Additional unit tests will be added in a separate PR after build paths are resolved properly.

**Pull request merged**